internal/collection/collection_test.go

```diff
@@ -17,3 +17,43 @@ func TestCopyWithSkip(t *testing.T) {
 		assert.Equal(t, []string{"a", "b", "c", "d", "e"}, CopyWithSkip(c, "f"))
 	})
 }
+
+func TestMapDiff(t *testing.T) {
+	tests := map[string]struct {
+		a        map[string]int
+		b        map[string]int
+		expected map[string]int
+	}{
+		"Disjoint maps": {
+			a:        map[string]int{"a": 1, "b": 2},
+			b:        map[string]int{"c": 3, "d": 4},
+			expected: map[string]int{"a": 1, "b": 2},
+		},
+		"Partially overlapping maps": {
+			a:        map[string]int{"a": 1, "b": 2, "c": 3},
+			b:        map[string]int{"b": 2, "d": 4},
+			expected: map[string]int{"a": 1, "c": 3},
+		},
+		"Fully overlapping maps": {
+			a:        map[string]int{"a": 1, "b": 2},
+			b:        map[string]int{"a": 1, "b": 2},
+			expected: map[string]int{},
+		},
+		"Empty map a": {
+			a:        map[string]int{},
+			b:        map[string]int{"a": 1},
+			expected: map[string]int{},
+		},
+		"Empty map b": {
+			a:        map[string]int{"a": 1, "b": 2},
+			b:        map[string]int{},
+			expected: map[string]int{"a": 1, "b": 2},
+		},
```

Comment:
This makes me think this is not a diff operation between 2 maps, but a kind of a subtraction of A - B.

In a proper diff result I would expect both added and removed fields. Here we only surface remaining fields after a subtraction.